ethtxn/ethtxn.go

```diff
@@ -83,14 +92,52 @@ func NewTransaction(ctx context.Context, provider *ethrpc.Provider, txnRequest *
 		txnRequest.GasLimit = gasLimit
 	}
 
+	if txnRequest.To == nil && (txnRequest.Data == nil || len(txnRequest.Data) == 0) {
+		return nil, fmt.Errorf("ethtxn: contract creation txn request requires data field")
+	}
+
 	var rawTx *types.Transaction
-	if txnRequest.To == nil {
-		if txnRequest.Data == nil || len(txnRequest.Data) == 0 {
-			return nil, fmt.Errorf("ethtxn: contract creation txn request requires data field")
+	if txnRequest.GasTip != nil {
+		chainId, err := provider.ChainID(ctx)
+		if err != nil {
+			return nil, err
 		}
-		rawTx = types.NewContractCreation(txnRequest.Nonce.Uint64(), txnRequest.ETHValue, txnRequest.GasLimit, txnRequest.GasPrice, txnRequest.Data)
+
+		rawTx = types.NewTx(&types.DynamicFeeTx{
+			ChainID:   chainId,
+			To:        txnRequest.To,
+			Nonce:     txnRequest.Nonce.Uint64(),
+			Value:     txnRequest.ETHValue,
+			GasFeeCap: txnRequest.GasPrice,
+			GasTipCap: txnRequest.GasTip,
+			Data:      txnRequest.Data,
+			Gas:       txnRequest.GasLimit,
```

Comment:
Is it possible for the caller to also set the `AccessList` along with the `GasTip`? Because this will ignore the `AccessList` in that case.

Reply:
Good catch! thanks